src/Network/Http/Auth/Oauth.php

```diff
@@ -135,6 +152,64 @@ protected function _hmacSha1($request, $credentials)
     }
 
     /**
+     * Use RSA-SHA1 signing.
+     *
+     * This method is suitable for plain HTTP or HTTPS.
+     *
+     * @param \Cake\Network\Http\Request $request The request object.
+     * @param array $credentials Authentication credentials.
+     * @return string
+     */
+    protected function _rsaSha1($request, $credentials)
+    {
+        if (!function_exists('openssl_pkey_get_private')) {
+            throw new \RuntimeException('RSA-SHA1 signature method requires the OpenSSL extension.');
+        }
+
+        $nonce = isset($credentials['nonce']) ? $credentials['nonce'] : uniqid();
```

Comment:
I copied the nonce default creation from the `_hmacSha1` method. Should it be `uniqid(true)` in this method too? The `_plaintext` method also just has `uniqid()`.
Btw. we can also use `Security::randomBytes()` like in #8890. :wink: